src/Mvc/Mvc.ViewFeatures/src/Filters/AntiforgeryApplicationModelProvider.cs

```diff
@@ -0,0 +1,55 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System.Linq;
+using Microsoft.AspNetCore.Antiforgery;
+using Microsoft.AspNetCore.Mvc.Core.Filters;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationModels;
+
+internal sealed class AntiforgeryApplicationModelProvider(IOptions<MvcOptions> mvcOptions, ILogger<AntiforgeryMiddlewareAuthorizationFilter> logger) : IApplicationModelProvider
+{
+    private readonly MvcOptions _mvcOptions = mvcOptions.Value;
+    private readonly ILogger<AntiforgeryMiddlewareAuthorizationFilter> _logger = logger;
+
+    public int Order => -1000 + 10;
+
+    public void OnProvidersExecuting(ApplicationModelProviderContext context)
+    {
+        ArgumentNullException.ThrowIfNull(context);
+
+        if (!_mvcOptions.EnableEndpointRouting)
+        {
+            return;
+        }
+
+        foreach (var controllerModel in context.Result.Controllers)
+        {
+            if (HasValidAntiforgeryMetadata(controllerModel.Attributes))
+            {
+                controllerModel.Filters.Add(new AntiforgeryMiddlewareAuthorizationFilter(_logger));
```

Comment:
Is it possible to only create one instance of `AntiforgeryMiddlewareAuthorizationFilter` and add that instance to all actions?